src/sentry/release_health/release_monitor/metrics.py

```diff
@@ -0,0 +1,203 @@
+                #     limit=self.CHUNK_SIZE + 1,
+                #     offset=offset,
+                # )
+                release_key = resolve_tag_key(org_id, "release")
+                release_col = Column(release_key)
+                env_key = resolve_tag_key(org_id, "environment")
+                env_col = Column(env_key)
+                query = (
+                    Query(
+                        dataset=Dataset.Metrics.value,
+                        match=Entity(EntityKey.MetricsCounters.value),
+                        select=[
+                            Function("sum", [Column("value")], "sessions"),
+                            Column("project_id"),
+                            release_col,
+                            env_col,
+                        ],
+                        groupby=[
+                            Column("project_id"),
+                            release_col,
+                            env_col,
+                        ],
+                        where=[
+                            Condition(
+                                Column("timestamp"),
+                                Op.GTE,
+                                datetime.utcnow() - timedelta(hours=6),
+                            ),
+                            Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
+                            Condition(Column("org_id"), Op.EQ, org_id),
+                            Condition(Column("project_id"), Op.IN, project_ids),
+                            Condition(
+                                Column("metric_id"),
+                                Op.EQ,
+                                indexer.resolve(org_id, SessionMRI.SESSION.value),
+                            ),
+                        ],
+                        granularity=Granularity(21600),
+                        orderby=[
+                            OrderBy(Column("project_id"), Direction.ASC),
+                            OrderBy(release_col, Direction.ASC),
+                            OrderBy(env_col, Direction.ASC),
+                        ],
+                    )
+                    .set_limit(self.CHUNK_SIZE + 1)
+                    .set_offset(offset)
+                )
+
+                with metrics.timer("release_monitor.fetch_project_release_health_totals.query"):
+                    # data = get_series(project_ids, query)
+                    data = raw_snql_query(
+                        query, referrer="release_monitor.fetch_projects_with_recent_sessions"
+                    )["data"]
+                    count = len(data)
+                    more_results = count > self.CHUNK_SIZE
+                    offset += self.CHUNK_SIZE
+
+                    if more_results:
+                        data = data[:-1]
+
+                    for row in data:
+                        env_name = indexer.reverse_resolve(row[env_key])
+                        release_name = indexer.reverse_resolve(row[release_key])
```

Comment:
It would probably be better if we had `bulk_reverse_resolve` here, and I could just grab all the unique things to resolve and do them in one go.